pkg/proxy/ipvs/proxier.go

```diff
@@ -1845,6 +1844,17 @@ func (proxier *Proxier) createAndLinkeKubeChain() {
 		}
 	}
 
+	// Handle traffic that loops back to the originator with SNAT.
+	// The rules should go before the kubernetes-specific ones
+	if !proxier.ipsetList[kubeLoopBackIPSet].isEmpty() {
```

Comment:
I don't think this would work because this is before we actually insert entries into the ipset https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/ipvs/proxier.go#L1164-L1169

Reply:
the test are passing so it has to be working :sweat_smile: 
but yeah, it seems it can skip one cycle, I think it works because the ipset will have always the coredns endpoints so once it's written it's always going to be there

Reply:
Yeah it may be "working" but I don't think this is the correct solution. On line 1081, we reset the ipset entries, but this happens after `createAndLinkeKubeChain` so here we are checking against the ipset from the previous sync.

Reply:
right, just was trying to understand why it worked, definitively we can not move ahead with this,.
I'll change it

Reply:
yeah I'm curious too, I think we can assume it's from the previous sync but not 100% sure  :thinking: